token/src/state.rs

```diff
@@ -396,6 +415,31 @@ impl State {
         Ok(())
     }
 
+    /// Processes a [BurnAccount](enum.TokenInstruction.html) instruction.
+    pub fn process_burn_account(program_id: &Pubkey, accounts: &[AccountInfo]) -> ProgramResult {
+        let account_info_iter = &mut accounts.iter();
+        let source_account_info = next_account_info(account_info_iter)?;
+        let dest_account_info = next_account_info(account_info_iter)?;
+        let authority_info = next_account_info(account_info_iter)?;
+
+        let mut source_data = source_account_info.data.borrow_mut();
+        let source_account: &mut Account = Self::unpack(&mut source_data)?;
+
+        Self::validate_owner(
+            program_id,
+            &source_account.owner,
+            authority_info,
+            account_info_iter.as_slice(),
+        )?;
+
+        **dest_account_info.lamports.borrow_mut() += source_account_info.lamports();
+        **source_account_info.lamports.borrow_mut() = 0;
+        source_account.amount = 0;
+        source_account.is_initialized = false;
```

Comment:
Why set `is_initialized` to false?

Reply:
Good measure, that account is no longer valid and should be purged, but to make sure if its not.  I guess we could just leave it initialized at zero balance...

Reply:
yeah, IMO just leave it.  Less BPF opcodes to execute.   It'll get rent collected soon enough.